test-alerts.cpp

```diff
@@ -3,83 +3,115 @@
 #include "test/catch.hpp"
 #include "typewise-alert.h"
 #include "Test-alert.h"
+#include <string.h>
 
-/***Global variables*****/
 
-int Test_GUI = NO_OUTPUT;
 
-/************************/
-
-TEST_CASE("infers the breach according to limits") 
+TEST_CASE("infers the breach according to limits: EMAIL testing") 
 {
-	int idx = 0;
-	Test_Parameters_st test_param[5]=
+	Email_Test_Buffer_st email_validate_buffer;
+	int email_test_idx = 0;
+	EMAIL_Test_Parameters_st test_param[2]=
 	{
+		
 		{
-			TO_CONTROLLER,
+			TO_EMAIL,
 			{
-				PASSIVE_COOLING,
+				MED_ACTIVE_COOLING,
 				"xxxx",
 			},
 			52,
-			CONTROLLER_OUTPUT,
-			0x00
+			"Hi, the temperature is too high",
+			"a.b@c.com"
 		},
 		{
-			TO_CONSOLE,
+			TO_EMAIL,
 			{
-				HI_ACTIVE_COOLING,
+				MED_ACTIVE_COOLING,
 				"xxxx",
 			},
-			-10,
-			CONSOLE_OUTPUT,
-			0x00
-		},
+			-55,
+			"Hi, the temperature is too low",
+			"a.b@c.com"
+		}
+	};
+	for(email_test_idx =0;email_test_idx<(sizeof(test_param)/sizeof(test_param[0]));email_test_idx++)
+	{
+		checkAndAlert(test_param[email_test_idx].altr ,test_param[email_test_idx].bat_ch,test_param[email_test_idx].temp );
+		email_validate_buffer = TEST_emailCodeMock(0,0,CHECK);
+		REQUIRE( strcmp(test_param[email_test_idx].expected_email_add , email_validate_buffer.address) == 0);
```

Comment:
These REQUIRE functions are testing the mock functionality. That's why they are not strong enough to catch issues in `checkAndAlert`, as mentioned in the previous comment.